test/units/TEST-75-RESOLVED.sh

```diff
@@ -1084,6 +1084,47 @@ testcase_13_varlink_subscribe_dns_configuration() {
         <(jq -cr --seq  '.configuration[] | select(.ifname == "dns0" and .servers != null and .searchDomains != null) | {"dns0":{servers: [.servers[] | .address], domains: [.searchDomains[] | .name]}}' "$tmpfile")
 }
 
+# Test RefuseRecordTypes
+testcase_14_refuse_record_types() {
+    # shellcheck disable=SC2317
+    cleanup() {
+        rm -f /run/systemd/resolved.conf.d/refuserecords.conf
+        if [[ -e /etc/resolv.conf.bak ]]; then
+            rm -f /etc/resolv.conf
+            mv /etc/resolv.conf.bak /etc/resolv.conf
+        fi
+        restart_resolved
+    }
+    trap cleanup RETURN ERR
+
+    mkdir -p /run/systemd/resolved.conf.d
+    {
+        echo "[Resolve]"
+        echo "RefuseRecordTypes=AAAA SRV TXT"
+    } >/run/systemd/resolved.conf.d/refuserecords.conf
+    if [[ -e /etc/resolv.conf ]]; then
+        mv /etc/resolv.conf /etc/resolv.conf.bak
+    fi
+    ln -svf /run/systemd/resolve/stub-resolv.conf /etc/resolv.conf
+    systemctl reload systemd-resolved.service
+    # disable_ipv6 is necessary do refuse AAAA
+    disable_ipv6
+    run dig localhost -t AAAA
+    grep -qF "status: REFUSED" "$RUN_OUT"
+
+    run dig localhost -t SRV
+    grep -qF "status: REFUSED" "$RUN_OUT"
+
+    run dig localhost -t TXT
+    grep -qF "status: REFUSED" "$RUN_OUT"
+
+    run dig localhost -t A
+    grep -qF "status: NOERROR" "$RUN_OUT"
+
+    run resolvectl query localhost5
```

Comment:
Please add more test cases with `resolvectl query --type=AAAA`, `SRV`, `TXT`, and `A`.

Reply:
added the `resolvectl query --type` test cases for AAAA, SRV, TXT and A.